modules/tensor_mechanics/src/materials/ComputeCreepPlasticityStress.C

```diff
@@ -185,8 +204,32 @@ ComputeCreepPlasticityStress::updateQpState(RankTwoTensor & elastic_strain_incre
       A22 += a * A12;
       rhs2 += a * rhs1;
 
-      const Real x2 = rhs2 / A22;
-      const Real x1 = (rhs1 - A12 * x2) / A11;
+      Real x2 = rhs2 / A22;
+      Real x1 = (rhs1 - A12 * x2) / A11;
+
+      while (effective_creep_strain_increment - x1 < 0)
+        x1 *= 0.5;
+      while (effective_plastic_strain_increment - x2 < 0)
+        x2 *= 0.5;
+
+      while (effective_creep_strain_increment - x1 > max_effective_creep_strain_increment)
+        x1 *= 0.5;
+      while (effective_plastic_strain_increment - x2 > max_effective_plastic_strain_increment)
+        x2 *= 0.5;
+
```

Comment:
This type of logic can be included or leveraged within a common, utility object (i.e. `NestedSolve`). That can be nice future work (not this PR).